crates/vqueues/src/scheduler/eligible.rs

```diff
@@ -149,17 +149,14 @@ impl EligibilityTracker {
                 State::NeedsPoll => {
                     let config = cache.config_pool().find(&qstate.qid.parent);
                     // update the state based on eligibility.
-                    match qstate
-                        .poll_eligibility(storage, SchedulerClock.now_millis(), meta, config)?
-                        .as_compact()
-                    {
+                    match qstate.poll_eligibility(storage, meta, config)?.as_compact() {
                         Eligibility::Eligible => {
                             *current_state = State::Ready;
                             return Ok(Some(handle));
                         }
                         Eligibility::EligibleAt(ts) => {
-                            let instant = SchedulerClock.millis_to_future_instant(ts);
-                            let timer_key = self.delayed_eligibility.insert_at(handle, instant);
+                            let duration = ts.duration_since(SchedulerClock.now_millis());
+                            let timer_key = self.delayed_eligibility.insert(handle, duration);
```

Comment:
One of the problems that I thought we would be solving with the addition of the `SchedulerClock` abstraction was that using durations can be problematic if there is a delay between calculating the duration (wake up time) and registering it at the delay queue. If there is a noticeable gap, then we can be woken up later. But I guess this problem gets dwarfed by Tokio's clock drift problem? 

So it might now happen that we sometimes wake up a little bit too early or too late depending on the clock drift and possible delays between calculating the wake up duration and inserting it into the `DelayQueue`?